cilium/CFP-26135-default-egress-allow.md

```diff
@@ -0,0 +1,198 @@
+
+
+## Motivation
+
+In cloud provider environments, it is often necessary to enforce firewall policies to restrict access to certain infrastructure components. Historically, before the emergence of eBPF, firewall rules were enforced using IPTables. Cilium primarily utilizes eBPF for policy enforcement. In eBPF host routing mode, Cilium bypasses IPtable rules and rules can be enforced only via eBPF. Also in certain scenarios (eg: L7), cilium prepends some IPTable rules by default which conflicts with cloud provider specific rules. Disabling the "prepend-iptables-chains" option is also not a viable solution as disabling prepend means Cilium can no longer guarantee that its policies are enforced.
+
+This proposal aims to bring the management of cloud provider specific policies within Cilium. The primary goal of this proposal is to integrate the cloud provider-specific policies into the Cilium framework. By doing so, we aim to address potential conflicts and other concerns associated with managing these policies across multiple components.
+
+## Goals
+
+- Block egress traffic to certain endpoints without blocking all
+- Behavior of existing network policy or cilium network policy should not be affected
+
+## Non Goals
+ - Ingress Restriction without blocking all
+ - Applying this policy for specific endpoints
+ - Complete Network Policy support with all features
+
+## Proposal
+
+### Overview
+The proposal will talk about preferred option and also about other options that aims to solve the problem by allowing cloud provider to specify the intent of blocking egress to certain endpoints without blocking all. These deny rules can be plumbed either via iptables or bpf policy rules. IPtables rules would not work with ebpf host routing mode. This proposal considers to solve this via bpf policy rules as it will work for both legacy host routing and ebpf host routing modes. It will also ensure that the behavior of existing network policies or Cilium network policies is not regressed.
+
+### Option 1 (CommandLine Flag)
+The decision to pass input as command line was taken based on what is more appropriate for this scenario. There is a section below that describes the reason for going with this option.
+ 
+A new command line flag, --deny-tuples=[ip:port:protocol] could be added as stringSlice. Cloud providers would customize this cilium config to specify apriori that traffic from Cilium-managed Pods towards specific address+port+protocol combinations should be blocked. This flag would effectively act independently of all of the NP, CNP, CCNP resources and enforce a deny for such tuples regardless of the user-defined policy. The deny tuples from cilium config are sanitized and converted to cilium egress deny policy rules. Then these rules should get added to policy map of all endpoints. Several options are considered to implement this and the following section would talk about each option
```

Comment:
nit, minor wording about what the proposal is - it's easier for reviewers to understand that this is the concrete plan assuming that this option is taken. Otherwise this is a really good summary of this option.

```suggestion
Add a new command line flag, --deny-tuples=[ip:port:protocol] as stringSlice. Cloud providers would customize this cilium config to specify apriori that traffic from Cilium-managed Pods towards specific address+port+protocol combinations should be blocked. This flag would effectively act independently of all of the NP, CNP, CCNP resources and enforce a deny for such tuples regardless of the user-defined policy. The deny tuples from cilium config are sanitized and converted to cilium egress deny policy rules. Then these rules should get added to policy map of all endpoints. 
```